TrafficSimulator/Assets/Utilities/CarSpawner.cs

```diff
@@ -165,114 +233,151 @@ private void CalculateLaneIndexes()
+        private void SpawnCars()
         {
-            LaneNode curr = lane.StartNode;
-            List<float> sections = new List<float>();
-            float sectionLength = 0;
+            int carsToSpawn = TotalCars;
 
-            while (curr != null)
+            // Return if there are no cars to spawn
+            if (carsToSpawn == 0)
             {
-                // Check if the current node is an intersection or if the next node is null to determine the end of a section
-                if(curr.RoadNode.IsIntersection())
-                {   
-                    sections.Add(sectionLength);
-                    sectionLength = 0;
-                    curr = curr.Next;
-                } 
-                else
+                Debug.Log("No cars to spawn");
+                return;
+            }
+
+            // Calculate cars to spawn per lane
+            List<CarsForLane> allLanes = _mode == SpawnMode.Total ? T_CalculateCarsPerLane(carsToSpawn) : R_CalculateCarsPerLane();
+
+            // Loop through all lanes and spawn their cars
+            foreach (CarsForLane lane in allLanes)
+            {
+                for (int i = 0; i < lane.Capacity - lane.CapacityLeft; i++)
                 {
-                    sectionLength += curr.DistanceToPrevNode;
+                    if(_carCounter >= TotalCars)
+                        return;
+                    
+                    _laneNodeCurrent = lane.SpawnableLaneNodes[i].Node;
+                    
+                    SpawnCar(0);
+                    _carCounter++;
                 }
-                
-                curr = curr.Next;
             }
-            
-            if(!lane.StartNode.Last.Prev.RoadNode.IsIntersection())
-                sections.Add(sectionLength);
-           
-            return sections;
         }
 
         /// <summary>Checks multiple conditions to determine if a car is able to spawn on node</summary>
         private bool IsCarSpawnable(LaneNode node)
         {
-            bool isThreeWayIntersection = node.RoadNode.Type == RoadNodeType.End && (node.Next?.IsIntersection() == true || node.Prev?.IsIntersection() == true);
-            return !(node.RoadNode.IsIntersection() || node.RoadNode.Type == RoadNodeType.JunctionEdge || node == null || node.Next == null || node.HasVehicle()) && !isThreeWayIntersection && !node.RoadNode.IsNavigationNode;
+            if(node == null)
+                return false;
+                
+            bool isThreeWayIntersection = true;
+            bool temp = false;
+
+            isThreeWayIntersection = node.RoadNode.Type == RoadNodeType.End && (node.Next?.IsIntersection() == true || node.Prev?.IsIntersection() == true);
+            temp = !(node.RoadNode.IsIntersection() || node.RoadNode.Type == RoadNodeType.JunctionEdge || node == null || node.Next == null || node.HasVehicle()) && !isThreeWayIntersection && !node.RoadNode.IsNavigationNode;
```

Comment:
No need for this variable, especially since it is called `temp`. Just return the condition and add a comment explaining the cases where the car is/isn't spawnable